src/credentials/providers/ec2CredentialsProvider.ts

```diff
@@ -1,27 +1,56 @@
 /*!
- * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import * as AWS from 'aws-sdk'
-import { CredentialType } from '../../shared/telemetry/telemetry.gen'
+import { Credentials, EC2MetadataCredentials, MetadataService } from 'aws-sdk'
 import { getStringHash } from '../../shared/utilities/textUtilities'
-import { CredentialsProvider, CredentialsProviderType ,CredentialsId } from './credentials'
-
+import { getLogger } from '../../shared/logger'
+import { CredentialType } from '../../shared/telemetry/telemetry.gen'
+import { CredentialsId, CredentialsProvider, CredentialsProviderType } from './credentials'
 
 /**
  * Credentials received from EC2 metadata service.
  *
  * @see CredentialsProviderType
  */
 export class Ec2CredentialsProvider implements CredentialsProvider {
-    public constructor(private token: string) {
+    private static readonly METADATA_SERVICE_TIMEOUT: number = 500
+
+    private credentials: EC2MetadataCredentials | undefined
+    private region: string | undefined
+    private available: boolean | undefined
+
+    public constructor(private metadata: MetadataService = Ec2CredentialsProvider.makeMetadataService()) {}
```

Comment:
We're also making a metadata service call for Cloud9 (region for metrics and CN). Can we converge the call logic?

Reply:
Created a common client for both of these flows to use - we can expand it if needed over time.